### PR TITLE
Refactor HttpEmitterConfig and ParametrizedUriEmitterConfig

### PR DESCRIPTION
**Pull request by leventov**
- Instead of subclassing `ParametrizedUriEmitterConfig` from `HttpEmitterConfig.Builder`, extract `BaseHttpEmittingConfig` superclass from `HttpEmitterConfig` and make it a nested field of `ParametrizedUriEmitterConfig`. @isimonenko have chosen to go the original way to keep `ParametrizedUriEmitterConfig` "flat", but I see nothing bad about have another level of nesting. In fact it more accurately models the config relations, and makes it more evident that Parametrized uses HttpPostEmitter under the hood. So instead of
```
xxx.parametrized.recipientBaseUrlPattern=...
xxx.parametrized.basicAuthentication=...
xxx.parametrized.batchingStrategy=...
```
it will be
```
xxx.parametrized.recipientBaseUrlPattern=...
xxx.parametrized.httpEmitting.basicAuthentication=...
xxx.parametrized.httpEmitting.batchingStrategy=...
```

 - Remove deprecated constructors in HttpEmitterConfig

### CODE REVIEW

**Comment by drcrallen** [comment]
If I recall part of the flat config was to facilitate migration and make sure settings weren't dropped, but @isimonenko should have more info.

File: src/main/java/com/metamx/emitter/core/HttpEmitterConfig.java [hunk 1]
```diff
@@ -17,258 +17,47 @@
...
-  {
-    return maxBufferSize;
-  }
-
-  public ContentEncoding getContentEncoding() {
-    return contentEncoding;
-  }
-
-  public static class Builder
+  public static class Builder extends HttpEmitterConfig
```
Comment (isimonenko) [inline src/main/java/com/metamx/emitter/core/HttpEmitterConfig.java]: extending HttpEmitterConfig is good to avoid code duplication but can lead to errors when somebody forget to call `build()` in the end and it won't be caught by compiler.

File: pom.xml [hunk 2]
```diff
@@ -25,7 +25,7 @@
     </parent>
 
     <artifactId>emitter</artifactId>
-    <version>0.6.0-rc2-SNAPSHOT</version>
+    <version>0.6.0-rc3-SNAPSHOT</version>
```
Comment (isimonenko) [inline pom.xml]: please remove rc* suffix before merge

Comment (leventov) [inline pom.xml]: The project is in fact on RC stage, so I don't see the point. OTOH want this to be in master already, to create other PRs based on this branch. And I don't want to create commits into this `refactor-http-and-parametrized-emitter-configs` feature branch, because when in the end it will be merged into master, it will contain a lot of unrelated changes.

File: src/main/java/com/metamx/emitter/core/HttpEmitterConfig.java (hunk 1 above)
Comment (leventov) [inline src/main/java/com/metamx/emitter/core/HttpEmitterConfig.java]: I think it's OK since 1) this Builder is used only in tests 2) actually would hardly be a problem if builder is used instead of a config object, the difference between them is that builder is mutable.
